aea/protocols/generator.py

```diff
@@ -0,0 +1,505 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+"""This module contains the protocol generator."""
+
+import argparse
+# import inspect
+import os
+import re
+from os import path
+from pathlib import Path
+import yaml
+from typing import Any, Dict, List, Set
+
+# CUR_PATH = os.path.dirname(inspect.getfile(inspect.currentframe()))
+# PWD = os.path.join(CUR_PATH, "..")
+PWD = ".."
+DEFAULT_TYPES = ["int", "float", "bool", "str", "bytes", "list", "dict", "tuple", "set"]
+MESSAGE_IMPORT = "from aea.protocols.base import Message"
+INIT_FILE_NAME = "__init__.py"
+MESSAGE_FILE_NAME = "message.py"
+SERIALIZATION_FILE_NAME = "serialization.py"
+PROTOCOL_FILE_NAME = "protocol.yaml"
+
+
+def to_snake_case(text):
+    """Convert a text in a CamelCase format into the snake_case format."""
+    s1 = re.sub('(.)([A-Z][a-z]+)', r'\1_\2', text)
+    return re.sub('([a-z0-9])([A-Z])', r'\1_\2', s1).lower()
+
+
+class ProtocolSpecificationParseError(Exception):
+    """Exception for parsing."""
+
+    pass
+
+
+class ProtocolTemplate:
```

Comment:
Can this inherit from ProtocolConfig? It seems that would avoid some duplication and make sense